src/utils/test/assertions.ts

```diff
@@ -0,0 +1,103 @@
+import { Template } from "aws-cdk-lib/assertions";
+import type { Resource } from "aws-cdk-lib/assertions/lib/private/template";
+import { TagKeys, TrackingTag } from "../../constants";
+import type { AppIdentity, GuStack } from "../../constructs/core";
+
+interface Tag {
+  Key: string;
+  Value: unknown;
+  PropagateAtLaunch?: boolean;
+}
+
+interface GuTagMatcherProps {
+  appIdentity?: AppIdentity;
+  additionalTags?: Tag[];
+  propagateAtLaunch?: boolean;
+}
+
+function sortTagsByKey(tags: Tag[]): Tag[] {
+  return [...tags].sort((first, second) => {
+    if (first.Key.toLowerCase() < second.Key.toLowerCase()) {
+      return -1;
+    }
+    if (first.Key.toLowerCase() > second.Key.toLowerCase()) {
+      return 1;
+    }
+    return 0;
+  });
+}
+
+export class GuTemplate {
```

Comment:
This class is a replacement for `test/jest.ts` and `test/synthed-stack.ts`. The aim is to provide a similar API to the [assertions model offered by AWS CDK v2](https://docs.aws.amazon.com/cdk/v2/guide/testing.html).

The API of `GuTemplate` isn't quite the same as `Template`: `new. GuTemplate(stack)` vs `Template.fromStack(stack)`. Happy to match the API if desired.